### PR TITLE
Rainbowcrack: init at 1.8

### PR DESCRIPTION
**Pull request by Tochiaha**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/fi/firewalk/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  buildInputs = [ libnet libpcap libdnet ];
+
+  meta = with lib; {
+    description = "Gateway ACL scanner";
+    homepage = "http://packetfactory.openwall.net/projects/firewalk/";
+    platforms = platforms.linux;
+    license = licenses.bsd2;
+    maintainers = with maintainers; [  ];
```
Comment (Lurkki14) [inline pkgs/by-name/fi/firewalk/package.nix:25]: You need to add yourself as a maintainer

File: pkgs/tools/security/rainbowcrack/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp -r * $out/bin
+  '';
+
+  meta = with pkgs.lib; {
+    description = "Rainbowcrack packaging for Kali Linux";
+    homepage = "https://gitlab.com/kalilinux/packages/rainbowcrack";
+    maintainers = with maintainers; [ ];
```
Comment (Lurkki14) [inline pkgs/tools/security/rainbowcrack/default.nix:29]: Same here

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/fi/firewalk/package.nix (hunk 1 above)
Comment (h7x4) [reply pkgs/by-name/fi/firewalk/package.nix:25]: Well, it's not strictly obligatory. It's written as an attribtute that [*should be set*](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes), but if you are very against it for some reason, it's okay.

File: pkgs/tools/security/rainbowcrack/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  passthru = {
+      NIXPKGS_ALLOW_UNFREE = "1";
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp -r * $out/bin
+  '';
```
Comment (h7x4) [inline pkgs/tools/security/rainbowcrack/default.nix:24]:
```suggestion
  installPhase = ''
    runHook preInstall
    
    mkdir -p $out/bin
    cp -r * $out/bin
    
    runHook postInstall
  '';
```

File: pkgs/tools/security/rainbowcrack/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    group = "kalilinux";
+    owner = "packages";
+    repo = "rainbowcrack";
+    rev = "kali/${version}";
+    hash = "sha256-y6jdFVQSO80YLLOitPAY8TVMUsi58B6+3bFoncwr+Qo=";
+  };
+
+  passthru = {
+      NIXPKGS_ALLOW_UNFREE = "1";
+  };
```
Comment (h7x4) [inline pkgs/tools/security/rainbowcrack/default.nix:19]: I'm not entirely sure what was the original purpose of this, but I'm pretty sure this is strictly illegal in nixpkgs.

File: pkgs/tools/security/rainbowcrack/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
+let
+  pkgs = import <nixpkgs> {  };
+
+in
+pkgs.stdenv.mkDerivation rec {
+  pname = "rainbowcrack";
+  version = "1.8-0kali1";
+
+  src = pkgs.fetchFromGitLab {
```
Comment (h7x4) [inline pkgs/tools/security/rainbowcrack/default.nix:9]:
```suggestion
{ lib, stdenNoCC, fetchFromGitLab }:

stdenvNoCC.mkDerivation rec {
  pname = "rainbowcrack";
  version = "1.8-0kali1";

  src = fetchFromGitLab {
```

File: pkgs/by-name/fi/firewalk/package.nix (hunk 1 above)
Comment (Lurkki14) [reply pkgs/by-name/fi/firewalk/package.nix:25]: PRs definitely have been held up from maintainers not being set, so I'm just recommending from that perspective

File: pkgs/tools/security/rainbowcrack/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  passthru = {
+      NIXPKGS_ALLOW_UNFREE = "1";
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp -r * $out/bin
+  '';
+
+  meta = with pkgs.lib; {
```
Comment (h7x4) [inline pkgs/tools/security/rainbowcrack/default.nix:26]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/security/rainbowcrack/default.nix [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    description = "Rainbowcrack packaging for Kali Linux";
```
Comment (h7x4) [inline pkgs/tools/security/rainbowcrack/default.nix:27]: This doesn't really describe what the package is very well. How about this?

```suggestion
    description = "Rainbow table generator to be used for password cracking";
```

File: pkgs/by-name/fi/firewalk/package.nix [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+, libpcap
+, libdnet
+}:
+
+stdenv.mkDerivation rec {
+  pname = "firewalk";
+  version = "5.0";
+
+  src = fetchurl {
+    url = "http://web.archive.org/web/20060925223218/http://www.packetfactory.net/firewalk/dist/firewalk-5.0.tgz";
```
Comment (h7x4) [inline pkgs/by-name/fi/firewalk/package.nix:14]: Have you verified that this builds? Page says 2003, if this works it's a miracle

File: pkgs/tools/security/rainbowcrack/default.nix [hunk 9]
```diff
@@ -0,0 +1,33 @@
+let
```
Comment (h7x4) [inline pkgs/tools/security/rainbowcrack/default.nix:1]: Why were one of these added to `by-name` and the other to `all-packages`?

**Review by h7x4 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/fi/firewalk/package.nix (hunk 1 above)
Comment (h7x4) [reply pkgs/by-name/fi/firewalk/package.nix:25]: Sure, that's true. But I still think it's important to know.

**Comment by h7x4** [comment]
For some reason, I'm not able to find the `rainbowcrack` package from toplevel `pkgs`.

But even so, I think the `rainbowcrack` derivation should be rewritten to build from source. The prebuilt version is probably dynamically linked against several libraries, and it's unnecessary to patchelf it when we have access to the source code at https://gitlab.com/kalilinux/packages/rainbowcrack.

**Pull request closed without merging**